pkg/deploy/controller/deployment/controller_test.go

```diff
@@ -651,6 +651,72 @@ func TestHandle_deployerPodDisappeared(t *testing.T) {
 	}
 }
 
+func expectMapContains(t *testing.T, exists, expected map[string]string, what string) {
+	if expected == nil {
+		return
+	}
+	for k, v := range expected {
+		value, ok := exists[k]
+		if ok && value != v {
+			t.Errorf("expected %s[%s]=%s, got %s", what, k, v, value)
+		} else if !ok {
+			t.Errorf("expected %s %s: not present", what, k)
+		}
+	}
+}
+
+func TestDeployerCustomLabelsAndAnnotations(t *testing.T) {
+	controller := &DeploymentController{
+		decodeConfig: func(deployment *kapi.ReplicationController) (*deployapi.DeploymentConfig, error) {
+			return deployutil.DecodeDeploymentConfig(deployment, api.Codec)
+		},
+		podClient: &podClientImpl{
+			createPodFunc: func(namespace string, pod *kapi.Pod) (*kapi.Pod, error) {
+				return pod, nil
+			},
+		},
+		makeContainer: func(strategy *deployapi.DeploymentStrategy) (*kapi.Container, error) {
+			return okContainer(), nil
+		},
+	}
+
+	testCases := []struct {
+		strategy     deployapi.DeploymentStrategy
+		labels       map[string]string
+		annotations  map[string]string
+		verifyLabels bool
+	}{
+		{deploytest.OkStrategy(), map[string]string{"label1": "value1"}, map[string]string{"annotation1": "value1"}, true},
```

Comment:
Sorry, just one more thing here... we're trying to clean up struct initializers for go vet. Can you please use explicit field names, like:

```go
{
  strategy: deploytest.OkStrategy(),
  labels: map[string]string{"label1": "value1"},
  annotations: map[string]string{"annotation1": "value1"},
  verifyLabels: true
},

```